data-prepper-api/src/main/java/com/amazon/dataprepper/model/buffer/Buffer.java

```diff
@@ -32,6 +32,14 @@
      */
     void write(T record, int timeoutInMillis) throws TimeoutException;
 
+    /**
+     * Atomically writes collection of records into the buffer
+     *
+     * @param records the collection of records to add
+     * @param timeoutInMillis how long to wait before giving up
+     */
```

Comment:
We should add `@throws PayloadSizeOverflowException ` to these Javadocs. (or whatever the name will be; see comments below).

Reply:
The way to write these would be:
```
@throws TimeoutException Unable to write to the buffer within the timeout
@throws SizeOverflowException The number of records exceeds the total capacity of the buffer. This cannot be retried.
@throws RuntimeException Other exceptions
```

This is an actual Javadoc syntax that expects an exception to come immediately after `@throws`.